core/src/test/scala/org/apache/spark/deploy/SparkSubmitUtilsSuite.scala

```diff
@@ -277,4 +279,35 @@ class SparkSubmitUtilsSuite extends SparkFunSuite with BeforeAndAfterAll {
         .exists(r.findFirstIn(_).isDefined), "resolution files should be cleaned")
     }
   }
+
+  test("SPARK-34624: should ignore non-jar dependencies") {
+    val main = MavenCoordinate("my.great.lib", "mylib", "0.1")
+    val dep = "my.great.dep:mydep:0.5"
```

Comment:
`0.5` is correct?

Reply:
The version number actually does not matter at all, it can be anything.

Reply:
If this doesn't matter, please use `0.1`. Otherwise, please document your description as a comment.

Reply:
Changed to `0.1`